# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContextLossHandler.h
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGLContextLossHandler.h
--- a/dom/canvas/WebGLContextLossHandler.h	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGLContextLossHandler.h	Wed Jan 08 22:19:14 2020 +0000
@@ -6,27 +6,18 @@
 #ifndef WEBGL_CONTEXT_LOSS_HANDLER_H_
 #define WEBGL_CONTEXT_LOSS_HANDLER_H_
 
+#include "mozilla/Atomics.h"
 #include "mozilla/WeakPtr.h"
-#include "nsCOMPtr.h"
-
-class nsIThread;
-class nsITimer;
+#include "mozilla/RefPtr.h"
+#include "nsThreadUtils.h"
 
 namespace mozilla {
 class WebGLContext;
 
 class WebGLContextLossHandler final
     : public SupportsWeakPtr<WebGLContextLossHandler> {
-  WebGLContext* const mWebGL;
-  const nsCOMPtr<nsITimer>
-      mTimer;          // If we don't hold a ref to the timer, it will think
-  bool mTimerPending;  // that it's been discarded, and be canceled 'for our
-  bool mShouldRunTimerAgain;  // convenience'.
-#ifdef DEBUG
-  nsISerialEventTarget* const mEventTarget;
-#endif
-
-  friend class WatchdogTimerEvent;
+  RefPtr<Runnable> mRunnable;
+  Atomic<bool> mTimerIsScheduled;
 
  public:
   MOZ_DECLARE_WEAKREFERENCE_TYPENAME(WebGLContextLossHandler)
@@ -35,9 +26,8 @@
   ~WebGLContextLossHandler();
 
   void RunTimer();
-
- private:
-  void TimerCallback();
+  // Allow future queueing of context loss timer.
+  void ClearTimer();
 };
 
 }  // namespace mozilla